wrapping up docs about rebase flow (#7401)
desktop/desktop: docs/technical/rebase-flow.md

 2. find the range of commits from `<upstream_oid>..<branch_oid>`
 3. generate a patch series of the contents, ready to rebase

### Detecting Conflicts

**TODO:** As part of detecting conflicts in [#6960](https://github.com/desktop/desktop/issues/6960)
there will be some words here eventually to help explain how the solution works.

## Warning about remote commits

If the target branch is tracking a remote branch and the user has enabled the

> _Show confirmation dialog before force pushing_ setting enabled, before starting

Reviewer: ```suggestion
_Show confirmation dialog before force pushing_ setting, Desktop
```

> the rebase Desktop will check if any commits exist in the range

Reviewer: ```suggestion
will check if any commits exist in the range
```

> `<upstream_oid>..<remote_branch_oid>`, where `<remote_branch_oid>` is the tip of

Reviewer: ```suggestion
`<upstream_oid>..<remote_branch_oid>` before starting the rebase, where `<remote_branch_oid>` is the tip of
```

the tracking branch.

This check is important to identify any remote commits that will be caught in

> this rebase, because this will require rewriting the history on the remote,

Reviewer: ```suggestion
this rebase, because their existence would require rewriting the history on the remote,
```

> which the user may not have permission to do based on branch protections.

Reviewer 2: ```suggestion
which the user may not have permission to do based on branch protections or may not want to do because of the impact it can have on other collaborators.
```

directory - both in files with conflict markers and files requiring a manual
resolution ("ours" or "theirs") - the rebase flow will determine what the next
rebase action should be.

After staging all the tracked files, if there are no changes in the index this
means the current commit is a no-op and the changes it contains are already
available in the base branch. This means the commit can be omitted by the
rebase by running `git rebase --skip`.

If there are changes in the index after staging all the changes, the rebase flow
will run `git rebase --continue` to indicate the current commit should be
updated and used in the rebased branch.

And like before, the rebase flow will monitor the output from this new rebase
command and report progress, until it either completes or encounters conflicts
again.

## Aborting the rebase

If the rebase encounters conflicts, the user has the opportunity to abort the
rebase if they feel uncomfortable with proceeding.

If the user has resolved conflicts before trying to abort, the rebase flow will
ask the user to confirm they wish to abort the rebase, as those changes will not
be available if they proceed.

## Force Push

> When the rebase is completed, the application will update it's list of rebased

Reviewer: ```suggestion
When the rebase is completed, the application will update its list of rebased
```
